apps/experiments/tests/test_data_migrations.py

```diff
@@ -0,0 +1,49 @@
+from apps.experiments.migration_utils import reconcile_connect_participants
+from apps.experiments.models import Participant
+from apps.utils.factories.experiment import ExperimentSessionFactory
+
+
+def test_reconcile_connect_participants(experiment):
+    team = experiment.team
+
+    # session1 should be ported to the participant of session2
+    session1 = ExperimentSessionFactory(
+        experiment=experiment, participant__platform="commcare_connect", participant__identifier="ABC123"
+    )
+    session2 = ExperimentSessionFactory(
+        team=team, experiment=experiment, participant__platform="commcare_connect", participant__identifier="abc123"
+    )
+
+    # session1's participant, different experiment. Should also be ported to session2's participant
+    session3 = ExperimentSessionFactory(team=team, participant=session1.participant)
+
+    # Another team, same setup. session4 should go to session5's participant
+    session4 = ExperimentSessionFactory(participant__platform="commcare_connect", participant__identifier="ABC123")
+    session5 = ExperimentSessionFactory(
+        team=session4.team, participant__platform="commcare_connect", participant__identifier="abc123"
+    )
+
+    # Totally different participant. This session's participant's identifier should be lowercased
+    session6 = ExperimentSessionFactory(
+        team=team, experiment=experiment, participant__platform="commcare_connect", participant__identifier="DEF456"
+    )
+
+    reconcile_connect_participants(Participant)
+
+    session1.refresh_from_db()
+    session2.refresh_from_db()
+    session3.refresh_from_db()
+    session4.refresh_from_db()
+    session5.refresh_from_db()
+    session6.refresh_from_db()
+
+    # Check that sessions 1, 2, and 3 are all linked to the same participant (the one with the lowercased identifier)
+    assert session1.participant_id == session2.participant_id == session3.participant_id
+    # Ensure the uppercase participant has been deleted
+    assert Participant.objects.filter(team=team, identifier="ABC123").exists() is False
+
+    assert session4.participant == session5.participant
```

Comment:
Also check that these are different from the other ones with the same ID in another team:
```suggestion
    assert session4.participant == session5.participant
    assert session4.participant != session1.participant
```

Reply:
Good point. See https://github.com/dimagi/open-chat-studio/pull/2237/commits/5373a9eee24c132bcd746163f881c0e6e36dcfea